src/wasi/WASI.cpp

```diff
@@ -41,14 +41,23 @@ static void* get_memory_pointer(Instance* instance, Value& value, size_t size)
     return memory->buffer() + offset;
 }
 
+template <class T, int maxSize>
 class TemporaryData {
 public:
     TemporaryData(size_t size)
     {
+        /* Check that more memory is requested than the maximum provided by malloc. */
+        if (size > ((~static_cast<size_t>(0)) / sizeof(T))) {
```

Comment:
Is this case possible?
It seems that this case would rarely happen, and not a intended operation.
So what about checking it simply with `ASSERT`?

Reply:
The size comes from WebAssembly, it can intentionally be a bad value (security). On 32 bit, the value can overflow and cause a crash.

Reply:
I see, then what about using `std::numeric_limits<size_t>` here?
It could be more intuitive I think

Reply:
Thank you! I have updated the patch